pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -1219,27 +1219,31 @@ protected CompletableFuture<TopicHashPositions> internalGetTopicHashPositionsAsy
                 });
     }
 
-    private String getBundleRange(String bundleName) {
-        NamespaceBundle nsBundle;
+    private CompletableFuture<String> getBundleRangeAsync(String bundleName) {
+        CompletableFuture<NamespaceBundle> future;
         if (BundleType.LARGEST.toString().equals(bundleName)) {
-            nsBundle = findLargestBundleWithTopics(namespaceName);
+            future = findLargestBundleWithTopicsAsync(namespaceName);
         } else if (BundleType.HOT.toString().equals(bundleName)) {
-            nsBundle = findHotBundle(namespaceName);
+            future = findHotBundleAsync(namespaceName);
         } else {
-            return bundleName;
-        }
-        if (nsBundle == null) {
-            return null;
+            return CompletableFuture.completedFuture(bundleName);
         }
-        return nsBundle.getBundleRange();
+        return future.thenApply(nsBundle -> {
+            if (nsBundle == null) {
+                return null;
+            }
```

Comment:
Can we just throw `RestException` here so that you don't need to check in https://github.com/apache/pulsar/pull/17402/files#diff-8696b1e2ba4ee355256e31b33ccc4f1cab77f1c6c79f7b4ab508b73d43d8ad55R1160-R1163

Reply:
fixed.